target/linux/ath79/image/common-tp-link.mk

```diff
@@ -83,3 +83,12 @@ define Device/tplink-loader-okli
   KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49 | \
 	loader-okli $(1) 12288
 endef
+
+define Device/tplink-qca-dsl
```

Comment:
I've just added a tplink-v2 definition to ath79:
https://github.com/openwrt/openwrt/commit/4897bf0ca5986663bae1086f12f7d76b860a7df3

You should be able to reuse that one instead of the tplink-qca-dsl here.

While at it, please check whether you need the "-s" for the KERNEL_INITRAMFS.

Reply:
I updated my commit and I can confirm you that it's working using the tplink-v2 definition (I tested both initramfs and sysupgrade image).

For what concerns the "-s" option, it's working?
Looking into https://github.com/openwrt/openwrt/blob/master/tools/firmware-utils/src/mktplinkfw2.c seems that the variable "strip_padding" is never used

Reply:
You were using the "-s" option for INITRAMFS, which I haven't seen elsewhere before (e.g. ramips does _not_ use it.) So I was interested in whether it was actually necessary, and since you now have removed it here (by invoking tplink-v2), it looks like you don't need it.

Reply:
Yes I don't need it.
However I was saying that the "-s" option doesn't do anything in mktplinkfw2.c... or am I missing something?